### ISSUE TITLE
chore: store node looses connection if database restarts

### ISSUE THREAD

**Issue by Ivansete-status** [waku-org/nwaku#2989]
### Background
After restarting the database, the store node doesn't connect again. Instead, it shows the following message until manually restarting the `nwaku` node:

> nim-waku-store  | ERR 2024-08-20 14:41:15.857+00:00 failed to insert message                   topics="waku archive" tid=1 file=archive.nim:108 error="could not put msg in messages table: error in runStmt: error in dbConnQueryPrepared calling sendQuery: failed pqsendQueryPrepared: no connection to the server\n"

**Comment by jakubgs** [comment]
Indeed, it appears only the insert code shows this behavior as we can see on this graph:

![image](https://github.com/user-attachments/assets/6aed9e15-16ae-4f61-8f31-e22ec7c7f8bd)

Because the query path continued to send statements the DB:

![image](https://github.com/user-attachments/assets/e030208d-e7f3-4e60-b4cd-e27b0dc06a39)
